earth2studio/utils/coords.py

```diff
@@ -255,7 +256,8 @@ def map_coords(
             indx = np.where(inc == outc[0])[0][0]
             inc_slice = inc[indx : indx + outc.shape[0]]
             if inc_slice.shape[0] == outc.shape[0] and np.all(inc_slice == outc):
-                x_slice = [slice(None)] * len(input_coords)
+                # Min here, to deal when coords have extra meta-data
+                x_slice = [slice(None)] * min([len(input_coords), x.ndim])
```

Comment:
Maybe also mark as TODO since eventually the `_lat`, `_lon` kludge will be removed